nav2-usability/scenario_code/lib/nested-routing/nested_routing_beamer.dart

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:beamer/beamer.dart';
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+// The state of the bottom navigation bar
+enum AppPage {
+  newBooks, // /books/new
+  allBooks, // /books/all
+  settings, // /settings
+}
+
+class BooksApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      title: 'Books App',
+      routerDelegate: BeamerRouterDelegate(
+        initialPath: '/books/new',
+        locationBuilder: SimpleLocationBuilder(
+          routes: {
+            '/*/*': (context) => HomeScreen(),
+          },
+        ),
+      ),
+      routeInformationParser: BeamerRouteInformationParser(),
+    );
+  }
+}
+
+class HomeScreen extends StatefulWidget {
+  @override
+  _HomeScreenState createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  final _innerBeamer = GlobalKey<BeamerState>();
+
+  @override
+  Widget build(BuildContext context) {
+    final beamerState = Beamer.of(context).state;
```

Comment:
```suggestion
    final beamerState = Beamer.of(context).currentLocation.state;
```

Reply:
@johnpryan It should be `Beamer.of(context).state;` this is the state above `currentLocation` (or any other stack within that `Beamer`)